common/lfdemod.c

```diff
@@ -1836,6 +1847,83 @@ int nrzRawDemod(uint8_t *dest, size_t *size, int *clk, int *invert, int *startId
     return 0;
 }
 
+size_t HitagPWMDemod(uint8_t *dest, size_t size, uint8_t *fchigh, uint8_t *fclow, uint32_t *startIdx, uint8_t *bitRange) {
+    printSignal();
+
+    if (g_debugMode == 2)
+        prnt("HitagWMDemod startIdx:%d low period:%d high period:%d", *startIdx, *fclow, *fchigh);
+    
+    size_t idx = 0, numBits = 0;
+
+    //find start of modulating data in trace
+    if (*startIdx == 0) {
+        idx = findModStart2(dest, size, *fchigh);
+        *startIdx = idx;
+        if (g_debugMode == 2)
+            prnt("detected start %d", idx);
+    }
+
+    uint8_t wave_min = 255;
+    uint8_t wave_max = 0;
+    uint8_t sub_idx = 0;
+    uint32_t idx_min = 0;
+    uint32_t idx_max = 0;
+    uint32_t last_wave_start = idx;
+    uint8_t meanCrossCount = 0;
+    for (; idx < size - 20; idx++, sub_idx++) {
+
+        if (dest[idx] < wave_min) {
+            wave_min = dest[idx];
+            idx_min = idx;
+        }
+        if (dest[idx] > wave_max) {
+            wave_max = dest[idx];
+            idx_max = idx;
+        }
+
+        uint32_t period = idx - last_wave_start;
```

Comment:
Would this work as a `int32_t`? As it currently stands, builds are failing under the Darwin compiler due to `period` being unsigned.

Reply:
What happens if "period" and "last_wave_start" are made size_t  (to match idx) rather then uint32_t ?
i.e. is size_t the same size as uint32_t ? (on some systems I believe size_t is 8 bytes, which would make it 64 bits, twice the size of a unit32_t.